Cargo.toml

```diff
@@ -11,6 +11,7 @@ members = [
   "prqlc/prqlc-parser",
   "prqlc/prqlc",
   "prqlc/prqlc/examples/compile-files", # An example
+  "lutra",
```

Comment:
@max-sixty lutra currnently does not compile on `wasm32-unknown-unknown`. That is partially because:
- there are (transitive) dependencies of connectorx that try to work with a terminal,
- rusqlite does not support `wasm32-unknown-unknown`, https://github.com/rusqlite/rusqlite/pull/1010, although it looks like it will soon

Is it feasible to disable compilation and testing of lutra on `wasm32-unknown-unknown`?
It is a crate that currently does not support WASM, so hiding these features behind conditionals feels dirty, as the resulting pruned crate would not have any functionality left.

Reply:
I'm talking about this: https://github.com/PRQL/prql/actions/runs/7654921007/job/20859910479?pr=4134

Reply:
> It is a crate that currently does not support WASM, so hiding these features behind conditionals feels dirty, as the resulting pruned crate would not have any functionality left.

I think that's fine; similar to how the `prqlc` CLI didn't compile on wasm — it instead had a main that immediately panics: https://github.com/PRQL/prql/blob/7c510444ce8fdbe4a53b6102095276f9ecfb3c1d/prqlc/prqlc/src/main.rs#L8-L9 (not part of the same crate but the comment applies to the binary)

Reply:
Remind me, why do we want to compile all binaries and libraries for WASM? Is it not ok to fail compilation, if what you are compiling cannot be used on WASM?

Reply:
We want to be able to run `cargo build` on WASM, without some modules failing — we'd prefer to specify what should be compiled in the rust code than in CLI args...

Reply:
But then we are building things that don't make sense? Compiling and linking binaries that are useless?

I don't think this is a good idea, so I won't do this myself. If you feel strongly about it and think that this repo is the best place for lutra to reside in, please do it yourself. Otherwise, this will go into a separate repo.

Reply:
@aljazerzen & I chatted briefly — I'm going to add onto this PR the "standard thing" of ignoring code under targets it can't compile to, to move this along.

If we have other approaches that work better, happy to change to those later! But let's not let devops get in the way of important work like this...

Reply:
Done